test/testother.cpp

```diff
@@ -4552,6 +4552,27 @@ class TestOther : public TestFixture {
               "    u.as_int = 0;\n"
               "}", true, false, false);
         ASSERT_EQUALS("", errout_str());
+
+        // Ticket #5115 "redundantAssignment when using a union"
+        check("#include <stdio.h>\n"
+              "typedef union{\n"
+              "    char as_char[4];\n"
+              "    int as_int;\n"
+              "} union_t;\n"
+              "void fn(char *data, int len) {\n"
```

Comment:
I don't see how this fn function would be related. it looks like it's copy pasted from above testcase.

also it's not your code to start with but the `#include <stdio.h>` should not be added in these unit tests because likely the code is not preprocessed.